### PR TITLE
workaround #378: generate a valid sarif file when no files are scanned

### PR DESCRIPTION
**Pull request by andyjones**
I use semgrep in github actions and upload the sarif file to Github. The job sometimes fails with a schema validation error when no files are scanned as described in #378 and #421 

This tweaks the output to generate a minimum sarif file that validates against the 2.1.0 json schema.

### Security

- [ ] Change has no security implications (otherwise, ping the security team)

### CODE REVIEW

**Comment by brendongo** [comment]
Sarif spec copied here matches https://github.com/oasis-tcs/sarif-spec/blob/master/Documents/CommitteeSpecifications/2.1.0/sarif-schema-2.1.0.json but looks like that spec doesn't match the spec published on https://docs.oasis-open.org/sarif/sarif/v2.1.0/os/sarif-v2.1.0-os.html#_Toc34317736 specifically the suppression object has `state` property in the former but is `status` in the later. Going to copy over the spec from https://docs.oasis-open.org/sarif/sarif/v2.1.0/cos02/schemas/sarif-schema-2.1.0.json

File: tests/unit/test_sarif.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+
+@pytest.fixture
+def sarif_schema():
+    with (Path(__file__).parent / "sarif-schema-2.1.0.json").open() as fd:
+        schema = json.load(fd)
+    return schema
+
+
+def test_rewrite_empty_sarif_file(tmp_path: Path, sarif_schema):
```
Comment (brendongo) [inline tests/unit/test_sarif.py:19]: Thanks for the test!

**Review by brendongo — APPROVED** [review]
Thanks for the contribution @andyjones !

**Comment by andyjones** [comment]
thanks @brendongo , I missed that! FYI in case it helps others, github use a slightly different/more lax schema when they validate the upload as well: https://github.com/github/codeql-action/commits/main/src/sarif_v2.1.0_schema.json

**Pull request merged**